pkg/apis/core/validation/validation.go

```diff
@@ -3398,11 +3399,14 @@ func validateTaintEffect(effect *core.TaintEffect, allowEmpty bool, fldPath *fie
 
 // validateOnlyAddedTolerations validates updated pod tolerations.
 func validateOnlyAddedTolerations(newTolerations []core.Toleration, oldTolerations []core.Toleration, fldPath *field.Path) field.ErrorList {
+	mergedNewTolerations := merge.DoTolerationsMerge(newTolerations, nil)
+	mergedOldTolerations := merge.DoTolerationsMerge(oldTolerations, nil)
+
```

Comment:
If we change the validation here, deleting existing tolerations that are dupped (some tolerations can be merged) will be allowed. This would be an api-change.

Reply:
Yes.. this change will have the effect of allowing the removal of dupped tolerations. 
I'm not sure if this will count as api-change, or if this kind of api-change is acceptable (dedupping is not the same as deleting, it might be acceptable?).
I just wanted to fix the bug mentioned in this PR, I haven't thought carefully about whether it's an api-change.